src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -102,12 +105,20 @@ impl Step for Std {
         let crates = std_crates_for_run_make(&run);
         let builder = run.builder;
 
-        // Force compilation of the standard library from source if the `library` is modified. This allows
-        // library team to compile the standard library without needing to compile the compiler with
-        // the `rust.download-rustc=true` option.
+        // Force compilation of the standard library from source if the `library` is modified. This
+        // allows library team to compile the standard library without needing to compile the
+        // compiler with the `rust.download-rustc=true` option.
         let force_recompile = builder.rust_info().is_managed_git_subrepository()
             && builder.download_rustc()
             && builder.config.last_modified_commit(&["library"], "download-rustc", true).is_none();
+        trace!(
+            ?force_recompile,
+            is_git_managed = builder.rust_info().is_managed_git_subrepository(),
+            download_rustc = builder.download_rustc(),
+            last_modified_commit =
+                ?builder.config.last_modified_commit(&["library"], "download-rustc", true),
+            "checking std force-recompile"
+        );
```

Comment:
Unrelated tracing diff, was mostly to help local debugging, I'd like to upstream some of these tracing bits separately (unrelated to r-a tests).